src/templates/finops-hub/modules/hub.bicep

```diff
@@ -1,305 +1,616 @@
+    userAssignedManagedIdentityResourceId: dataFactoryScriptsIdentity.id
+    userAssignedManagedIdentityPrincipalId: dataFactoryScriptsIdentity.properties.principalId
+    networkingOption: networkingOption
+    newScriptsSubnetResourceId: (networkingOption == 'Private') ? vnet.outputs.subnetResourceIds[1] : null
+  }
+}
+
+//------------------------------------------------------------------------------
+// Key Vault for storing secrets
+//------------------------------------------------------------------------------
+
+module keyVault 'keyVault.bicep' = {
+  name: 'keyVault'
+  params: {
+    hubName: hubName
+    uniqueSuffix: uniqueSuffix
+    location: location
+    tags: resourceTags
+    tagsByResource: tagsByResource
+    storageAccountName: storage.outputs.name
+    subnetResourceId: (networkingOption == 'PrivateWithExistingNetwork') ? subnetResourceId : (networkingOption == 'Private') ? vnet.outputs.subnetResourceIds[0] : ''
+    accessPolicies: [
+      {
+        objectId: dataFactory.identity.principalId
+        tenantId: subscription().tenantId
+        permissions: {
+          secrets: [
+            'get'
+          ]
+        }
+      }
+    ]
+    networkingOption: networkingOption
+    newsubnetResourceId: (networkingOption == 'Private') ? vnet.outputs.subnetResourceIds[0] : null
+    keyVaultPrivateDNSZoneName: keyVaultPrivateDNSZoneName
+    privateDNSZonesResourceGroupName: privateDNSZonesResourceGroupName
+    virtualNetworkName: (networkingOption == 'Private') ? vnet.name : null
+    }
+}
+
+//------------------------------------------------------------------------------
+// Virtual network for private access
+//------------------------------------------------------------------------------
+
+module vnet 'br/public:avm/res/network/virtual-network:0.1.8' = if (networkingOption == 'Private') {
+  name: networkName
+  params: {
+    name: networkName
+    addressPrefixes: [
+      networkAddressPrefix
+    ]
+    location: location
+    tags: union(tags, tagsByResource[?'Microsoft.Network/virtualNetworks'] ?? {})
+    subnets: [
+      {
+        name: networkSubnetName
+        addressPrefix: networkSubnetPrefix
+      }
+      {
+        name: scriptsSubnetName
+        addressPrefix: scriptsSubnetPrefix
+        serviceEndpoints: [
+          {
+            service: 'Microsoft.Storage'
+          }
+        ]
+        delegations: [
+          {
+            name: 'Microsoft.ContainerInstance.containerGroups'
+            properties: {
+              serviceName: 'Microsoft.ContainerInstance/containerGroups'
+            }
+          }
+        ]
+      }
+    ]
+  }
+}
+
+
+
+//------------------------------------------------------------------------------
+// Private Endpoints for ADF
+//------------------------------------------------------------------------------
+
+resource ADFprivateDNSZone 'Microsoft.Network/privateDnsZones@2020-06-01' existing = if(networkingOption == 'PrivateWithExistingNetwork') {

```

Comment:
nit: Our naming standards are to make these camel-case. Also note 3-character acronyms should be Pascal-cased (e.g., `Adf`).